docs/advanced-topics/entt.md

```diff
@@ -0,0 +1,271 @@
+# EnTT
+
+## Introduction
+
+[EnTT](https://github.com/skypjack/entt) is the [data-oriented](https://en.wikipedia.org/wiki/Data-oriented_design)
+entity component system (ECS) used by Minecraft: Bedrock Edition. It is used to store common data for entities, such
+as position and collision boxes, in discrete structs referred to as components.
+
+With each version release, more pieces of `Actor` data are moved from class members to dedicated components;
+fortunately, EnTT makes it possible to respond quickly to these changes and maintain up-to-date Minecraft version
+compatibility.
+
+For more information on EnTT, check out the highly informative and comprehensive
+[wiki](https://github.com/skypjack/entt/wiki) written by its [maintainer](https://github.com/skypjack).
+
+## Picking the Version
+
+The specific version of EnTT that is required for your Minecraft version will vary:
+
+| MC:BE Release     | EnTT Tag  | EnTT Commit Hash                           |
+|:------------------|:----------|:-------------------------------------------|
+| `1.20.70+`        | `v3.13.1` | `2909e7ab1f1e73a36f778319070695611e3fa47b` |
+| `1.20.50-1.20.6x` | N/A       | `62a13526c989f14eff348c28c061542ac7a16d45` |
+| `1.20.4x`         | N/A       | `717897052477515260bde3fd21fe987662666621` |
+
+## Including EnTT
+
+*This guide only addresses including EnTT for CMake based projects.*
+
+The CMake [FetchContent](https://cmake.org/cmake/help/latest/module/FetchContent.html) module can be used to download
+and include EnTT's CMake project from a git commit hash or tag:
+```CMake
+include(FetchContent)
+
+FetchContent_Declare(
+  EnTT
+  GIT_REPOSITORY https://github.com/skypjack/entt.git
+  GIT_TAG        2909e7ab1f1e73a36f778319070695611e3fa47b # v3.13.1
+)
+
+FetchContent_MakeAvailable(EnTT)
+target_link_libraries(MyTarget PRIVATE EnTT::EnTT)
+```
+
+Alternatively, if a package repository is configured in your environment, `find_package` can be used as a less verbose
+option. However, only official releases that are published to the repository will be available, not necessarily specific
+commits.
+```CMake
+find_package(EnTT 3.13.1 CONFIG REQUIRED)
+target_link_libraries(MyTarget PRIVATE EnTT::EnTT)
+```
+
+## Configuration
+
+Mojang has configured EnTT in multiple ways to suit the needs of Minecraft: Bedrock Edition. In order to have binary
```

Comment:
Maybe word the last bit as something in like "In order for mods to have binary...", this just makes it more clear what subject you're referring to after you were talking about the game